[Seng Zhen Hong] iP

IamZhenHong opened the pull request:
CharmBot
“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
FAST SUPER FAST to use
All you need to do is,


add your tasks.
let it manage your tasks for you 😉
And it is FREE!

Features:

 Managing tasks
 Managing deadlines (coming soon)
 Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Initial implementation of CharmBot with greeting and exit

Commit:
Enhance Charmbot to echo user commands and exit on bye

Commit:
Add text storage and display functionality to CharmBot

IamZhenHong committed:
Added mark and unmark functionality to Charmbot

IamZhenHong committed:
Added Todos,Events,Deadlines tracking for CharmBot

IamZhenHong committed:
Added Automated Test UI Testing

IamZhenHong committed:
Added exceptions to handle input errors

IamZhenHong committed:
Added error handling functionality to CharmBot

IamZhenHong committed:
Added support for deleting tasks from list

IamZhenHong committed:
Enabling saving and loading data with hard disk

IamZhenHong committed:
Merge branch 'branch-Level-7df into main

IamZhenHong committed:
Added date and time to task objects

dhlee03 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+        while (!command.equals("bye")) {
+            System.out.println("____________________________________________________________");
+            try {
+                if (command.equals("list")) {
```
The code seems to be too long for me. Try to divide each commands into different functions.

dhlee03 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+                LocalTime endTime = LocalTime.parse(to, DateTimeFormatter.ofPattern("HHmm"));
+                return new EventTask(description, startTime, endTime);
+            default:
+                return new ToDoTask(description);
```
Consider the case when the file is corrupted

dhlee03 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+
+class Task {
+    protected String description;
+    protected boolean isDone;
```
Nice boolean naming👍

dhlee03 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+        }
+        duke.saveTasksToFile(tasks);
+    
+        System.out.println("____________________________________________________________");
```
Maybe consider connecting this three println statements using +.

dhlee03 reviewed (commented):
Nice Code!. Keep working on it:)

yorklim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+import java.io.File;
```
Can consider ordering imports in lexical order

yorklim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("____________________________________________________________");
```
Can consider making linebreaks into a constant variable which can make code cleaner

yorklim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+        String command = scanner.nextLine();
+         // Initialize tasks list here
+    
+        while (!command.equals("bye")) {
```
Can consider breaking down this function into smaller functions to make it more readable for others, might be a little too long currently. Can break down into like two functions like run and commands

yorklim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+                scanner.close();
+            }
+        } catch (IOException e) {
+            System.out.println("Error loading tasks from file: " + e.getMessage());
```
Can consider making this into your own exception class and include the additional error messages as part of the new exception

yorklim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
```
Is there a reason why you added this. infront in task and not in Duke class? Can consider having a constant reference standard.

yorklim commented on src/main/java/Duke.java:
Good naming of functions and variables and all in all LGTM but just some minor nits

yorklim reviewed (commented):
LGTM just some minor nits to consider!

IamZhenHong committed:
Made the code more OOP

IamZhenHong committed:
Divided classes into packages

IamZhenHong committed:
Merge branch 'add-gradle-support'

IamZhenHong committed:
Set up gradle

IamZhenHong committed:
Added JUnit test for Ui and Task

IamZhenHong committed:
Package app as jar file

IamZhenHong committed:
Added JavaDoc commments

IamZhenHong committed:
Added find function

IamZhenHong committed:
Added checkstyle

IamZhenHong committed:
Added GUI

IamZhenHong committed:
Add assertion to ensure non-empty array in parseCommand method

This commit adds an assertion to the parseCommand method in the Parser class to ensure that the array 'parts' has at least one element before further processing. The added assertion enhances the robustness of the code by catching unexpected conditions early in development.

Changes made in this commit:
- Added assertion 'assert parts.length > 0;' to validate the length of the array 'parts'.

This assertion serves as a safeguard against potential runtime errors and improves the overall reliability of the command parsing logic.

IamZhenHong committed:
Merge pull request #2 from IamZhenHong/branch-A-Assertions

Add assertion to ensure non-empty array in parseCommand method

IamZhenHong committed:
Improved code quality

IamZhenHong committed:
Merge pull request #3 from IamZhenHong/branch-A-CodeQuality

Improved code quality

IamZhenHong committed:
Improved GUI

IamZhenHong committed:
Merge pull request #4 from IamZhenHong/GUI

G UI

IamZhenHong committed:
Update README.md

IamZhenHong committed:
Update README.md

IamZhenHong committed:
Update README.md

IamZhenHong committed:
Added Ui.png

IamZhenHong committed:
Merge branch 'master' of github.com:IamZhenHong/ip

IamZhenHong committed:
Update README.md

IamZhenHong committed:
Added welcome message
